### PR TITLE
Detect extension using `content-type` if magic byte check fails

### PR DESCRIPTION
**Pull request by kevva**
Hard to test this since `file-type` detects most commonly used file types.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -12,10 +12,27 @@ const makeDir = require('make-dir');
...
+const getExtFromMime = res => {
+	const header = res.headers['content-type'];
+
+	if (!header) {
+		return null;
+	}
+
+	const exts = extName.mime(header);
+
+	if (exts.length !== 1) {
```
Comment (silverwind) [inline index.js]: Did you mean `!exts.length`? This would fail to return a extension if two or more are returned.

Comment (kevva) [inline index.js]: Yes, that's intended. If it returns more than one, we can't reliably set the extension, IMO.

Comment (kevva) [inline index.js]: Look at https://github.com/jshttp/mime-types#mimeextensiontype for example. `zip` files can also have that `content-type` which in this case would result in an incorrect extension being set.

Comment (silverwind) [inline index.js]: Looking through `mime-db`, it seems it's returning the "right" extension first in most cases, but there are also cases like this:

````json
  "application/x-msdownload": {
    "source": "apache",
    "extensions": ["exe","dll","com","bat","msi"]
},
````

So I guess it's fine for those cases to fail.

**Pull request merged**